web/src/app/lists/FlatList.js

```diff
@@ -1,228 +1,225 @@
+ * function is provided. Other options are listed in the
+ * propTypes.
+ */
+export default function FlatList(props) {
+  const classes = useStyles()
+  const absURL = useSelector(absURLSelector)
 
-  onDragStart = () => {
-    // adds a little vibration if the browser supports it
-    if (window.navigator.vibrate) {
-      window.navigator.vibrate(100)
-    }
-  }
+  return props.onReorder ? renderDragAndDropList() : renderList()
 
-  onDragEnd = result => {
-    this.props.onReorder(
-      // result.draggableId, : removed this as per new reorderList function
-      result.source.index,
-      result.destination.index,
+  function renderDragAndDropList() {
+    return (
+      <DragDropContext onDragStart={onDragStart} onDragEnd={props.onReorder}>
+        <Droppable droppableId='droppable'>
+          {(provided, _) => (
+            <div ref={provided.innerRef} {...provided.droppableProps}>
+              {renderList(provided.placeholder)}
+            </div>
+          )}
+        </Droppable>
+      </DragDropContext>
     )
   }
 
-  renderItem(item, idx) {
-    let itemProps = {}
-    if (item.url) {
-      itemProps = {
-        component: Link,
-        to: this.props.absURL(item.url),
-        button: true,
-      }
-    }
+  function renderList(dragPlaceholder) {
+    const { onReorder, emptyMessage, headerNote, items, ...otherProps } = props
+
+    const subheader = headerNote ? (
+      <ListSubheader>{headerNote}</ListSubheader>
+    ) : null
 
     return (
-      <ListItem
-        key={idx}
-        {...itemProps}
-        style={{ width: '100%' }}
-        className={item.highlight ? this.props.classes.highlightedItem : null}
-      >
-        {item.icon && <ListItemIcon>{item.icon}</ListItemIcon>}
-        <ListItemText
-          primary={item.title}
-          secondary={item.subText}
-          secondaryTypographyProps={{ style: { whiteSpace: 'pre-line' } }}
-        />
-        {item.secondaryAction && (
-          <ListItemSecondaryAction>
-            {item.secondaryAction}
-          </ListItemSecondaryAction>
-        )}
-      </ListItem>
+      <List subheader={subheader} {...otherProps}>
+        {renderListItems()}
+
+        {/* Rendered if props.onReorder is specified */}
+        {dragPlaceholder}
+      </List>
     )
   }
 
-  renderItems() {
-    if (!this.props.items.length) {
-      return (
-        <ListItem>
-          <ListItemText
-            disableTypography
-            secondary={
-              <Typography variant='caption'>
-                {this.props.emptyMessage}
-              </Typography>
-            }
-          />
-        </ListItem>
-      )
+  /*
+   * Handles rendering either as empty message text,
+   * standard list items, list items that support
+   * drag and drop, or as a subheader list item.
+   */
+  function renderListItems() {
+    // render as empty message
+    if (!props.items.length) {
+      return <ListSubheader>{props.emptyMessage}</ListSubheader>
```

Comment:
Does this change how it's displayed?

Reply:
It shouldn't! Using this how the material-ui.com docs recommend